### ISSUE TITLE
StackPanels do not position controls properly.

### ISSUE THREAD

**Issue by atom0s** [NeoforceControls/XNA#7]
When adding stack panels, and items to stack panels they do not reposition properly.
I believe this is due to CalcLayout not being called when a new control is added. I partially fixed the issues adding the following to StackPanel.cs:

``` C#
      public override void Add(Control control)
      {
          base.Add(control);
          this.CalcLayout();
      }
      public override void Add(Control control, bool client)
      {
          base.Add(control, client);
          this.CalcLayout();
      }
```

**Comment by Grimston** [comment]
Found the correct way.

Call Refresh on the stack panel after adding all controls to it.

-------- Original message --------

**Comment by atom0s** [comment]
That works as well. But adding controls at runtime, it's a little inconvenient to have to call Refresh manually each time.

**Comment by Grimston** [comment]
Its only needed once at the end of adding to the stack panel.

I will add a auto refresh property to the stack panel tonight. To make things easier.

-------- Original message --------

**Comment by atom0s** [comment]
Any chance you are adding docking in the near future too? So that controls will auto-resize to their parent based on docking settings like normal WinForms has?

**Comment by Grimston** [comment]
Was thinking about that recently actually. Will look into it.

-------- Original message --------

**Comment by Grimston** [comment]
Fixed 8e29f36
